core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/visualization/scatterplot/ScatterplotOpExec.java

```diff
@@ -0,0 +1,44 @@
+package edu.uci.ics.texera.workflow.operators.visualization.scatterplot;
+
+import edu.uci.ics.texera.workflow.common.operators.map.MapOpExec;
+import edu.uci.ics.texera.workflow.common.tuple.Tuple;
+import edu.uci.ics.texera.workflow.common.tuple.schema.Attribute;
+import edu.uci.ics.texera.workflow.common.tuple.schema.Schema;
+import scala.Function1;
+import scala.Serializable;
+import java.util.*;
+
+
+/**
+ * Scatterplot operator to visualize the result as a scatterplot
+ *
+ */
+public class ScatterplotOpExec extends MapOpExec {
+    private final ScatterplotOpDesc opDesc;
+
+    public ScatterplotOpExec(ScatterplotOpDesc opDesc) {
+        this.opDesc = opDesc;
+        this.setMapFunc((Function1<Tuple, Tuple> & Serializable) this::processTuple);
+    }
+
+    public Tuple processTuple(Tuple t) {
+        List<Object> resultObjects = new ArrayList<>();
+
+        Schema resultSchema;
+        if(opDesc.isGeometric)
+            resultSchema = Schema.newBuilder().add(
+                    new Attribute("longitude", t.getSchema().getAttribute(opDesc.xColumn).getType()),
+                    new Attribute("latitude", t.getSchema().getAttribute(opDesc.yColumn).getType())
```

Comment:
in the Desc, you defined the output schema to be `xColumn` and `yColumn`. why are you using another schema here?

Reply:
because the value of the xColumn can be any name of the column chosen by the user for non-spatial scatterplot but I am fixing it for spatial scatterplot, I can use `xColumn` and `yColumn` to avoid confusion

Reply:
Remember we are detaching schema from tuples. So having different schema in operator output schema and tuple is not ideal.

Reply:
that will be handled later, whenever the detaching schema PR happens not now

Reply:
I still think you should unify the described schema and the actual schema in the tuple, even if right now schema is not detached from tuple yet. @maazadeeb is working on this and we don't want to creat more overhead to him: he might now know the operator logic, it will be hard for him to perform the detachment.